java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/IdleTimeoutTransceiverDecorator.java

```diff
@@ -126,24 +120,30 @@ public IdleTimeoutTransceiverDecorator(
         _sendHeartbeat = () -> connection.sendHeartbeat();
     }
 
-    public boolean isIdleCheckEnabled() {
+    boolean isIdleCheckEnabled() {
         return _idleCheckEnabled;
     }
 
-    public void enableIdleCheck() {
+    void enableIdleCheck() {
         if (!_idleCheckEnabled && _idleCheck != null) {
             rescheduleReadTimer();
             _idleCheckEnabled = true;
         }
     }
 
-    public void disableIdleCheck() {
+    void disableIdleCheck() {
         if (_idleCheckEnabled && _idleCheck != null) {
             cancelReadTimer();
             _idleCheckEnabled = false;
         }
     }
 
+    void scheduleHeartbeat() {
+        // Reschedule because the connection establishment may have already written to the
```

Comment:
Keep in mind that a "write" can occur during connection establishment. This pre-Active write schedules a heartbeat, but this heartbeat will be ignored if we attempt to send it before the ice connection is Active.